### PR TITLE
Issue 105 new commit

### PR DESCRIPTION
**Pull request by Amitbhave**
@sebbrudzinski You had suggested earlier, that not to pass those "previous" and "next" buttons across 2 classes, but when the PDF preview refreshes, it is handled in PDFPreviewRefresher.java; which has no access to those 2 buttons. So in my opinion, it won't be possible to achieve this without passing the buttons.

Please find the pull request and let me know your feedback.

### CODE REVIEW

File: OpenLaTeXStudio/Editor/src/main/java/latexstudio/editor/pdf/PDFDisplay.java [hunk 1]
```diff
@@ -1,96 +1,107 @@
...
+    private int selectedPage;
+    private int totalPages;
+
+    public static boolean isDisablePrevFlag() {
+        return disablePrevFlag;
+    }
+
+    public static boolean isDisableNextFlag() {
+        return disableNextFlag;
+    }
```
Comment (sebbrudzinski) [inline OpenLaTeXStudio/Editor/src/main/java/latexstudio/editor/pdf/PDFDisplay.java]: Those two methods shouldn't be static.

File: OpenLaTeXStudio/Editor/src/main/java/latexstudio/editor/PDFViewerTopComponent.java [hunk 2]
```diff
@@ -280,7 +286,16 @@ public void componentClosed() {
...
+        if(PDFDisplay.isDisableNextFlag()){
+            jButton2.setEnabled(false);
+        } else{
+            jButton2.setEnabled(true);
+        }
+        if(PDFDisplay.isDisablePrevFlag()){
+            jButton1.setEnabled(false);
+        } else{
+            jButton1.setEnabled(true);
+        }
```
Comment (sebbrudzinski) [inline OpenLaTeXStudio/Editor/src/main/java/latexstudio/editor/PDFViewerTopComponent.java]: We set buttons enabled/disabled in two places in this class and both places use different condition to do so. This is not good.

File: OpenLaTeXStudio/Editor/src/main/java/latexstudio/editor/pdf/PDFPreviewRefresher.java [hunk 3]
```diff
@@ -1,86 +1,102 @@
...
+        if(PDFDisplay.isDisableNextFlag()){
+            nextPageButton.setEnabled(false);
+        } else{
+            nextPageButton.setEnabled(true);
+        }
+        if(PDFDisplay.isDisablePrevFlag()){
+            previousPageButton.setEnabled(false);
+        } else{
+            previousPageButton.setEnabled(true);
+        }
```
Comment (sebbrudzinski) [inline OpenLaTeXStudio/Editor/src/main/java/latexstudio/editor/pdf/PDFPreviewRefresher.java]: One more place where we have the same logic coded. This should be defined in a single place.

**Comment by sebbrudzinski** [comment]
I could live with passing the button instances, but there are several other problems in the code that need to be resolved.

**Comment by Amitbhave** [comment]
Yes, I will take care of the problems in my next commit. Thanks for your feedback.

**Pull request closed without merging**